### ISSUE TITLE
Tags-Deselecting state tag shows state no longer valid error

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49504]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.38-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4988212
**Issue reported by:** Applause Internal Team

## Action Performed:
Pre-conditions: 

- In Android app, login as admin &create a workspace
- Invite a employee and as employee keep logged in mweb
- Enable tags for admin in app

1. Login old dot as admin of workspace
2. Tap settings --workspace-- tags
3. Enable "use multiple level of tags", verify " People must tag expense also enabled (if not enable it)
4. Tap import from spread sheet and select a depedent tag file attached below
5. Tap upload tags
6. In new dot, open workspace settings- tags
7. Disable region-- North+test
8. Open in mweb as employee, workspace chat
9. Submit a expense entering merchant
10. Open the expense created
11. Deselect the state tag

## Expected Result:
Deselecting state tag must show state tag missing error.
## Actual Result:
Deselecting state tag shows state no longer valid error.
## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a2d0a8e9-7c08-4a64-933d-a348c031d0e0

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
@sakluger  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by sakluger** [comment]
@IuliiaHerets I think you attached the wrong video, it doesn't match the reproduction steps. Could you please check?

**Comment by IuliiaHerets** [comment]
@sakluger Fixed it, sorry

**Comment by rezkiy37** [comment]
Hi, I am Michael (Mykhailo) from Callstack, an expert agency and I can work on this issue.

**Comment by sakluger** [comment]
Thanks @rezkiy37 for helping on this one!

**Comment by rezkiy37** [comment]
I am trying to reproduce the bug.

**Comment by rezkiy37** [comment]
@IuliiaHerets, I cannot see the attached file. 
> 4. Tap import from spread sheet and select a depedent tag file attached below

**Comment by rezkiy37** [comment]
I couldn't reproduce the bug on the latest `main`. I will try one more time.

<details><summary>Details</summary>
<p>

![Screenshot_20241001_163053_Chrome](https://github.com/user-attachments/assets/aa01abb3-f9ab-4a98-8199-8fce55637894)
![Screenshot_20241001_163159_New Expensify Dev](https://github.com/user-attachments/assets/2115c0e0-a321-4f51-9286-c9cae299f4d4)

</p>
</details>

**Comment by sakluger** [comment]
Thanks @rezkiy37. @IuliiaHerets are you still able to reproduce on the latest version?

**Comment by rezkiy37** [comment]
Not replicable again.

<details><summary>Video</summary>
<p>


https://github.com/user-attachments/assets/74a745bb-8829-427f-8895-fd44714e9355



</p>
</details>

**Comment by sakluger** [comment]
@IuliiaHerets are you still able to reproduce this one? I'd like your confirmation before closing the issue.

**Comment by IuliiaHerets** [comment]
@sakluger Tester still can reproduce the issue

https://github.com/user-attachments/assets/25019691-9fb4-4297-b495-e19c88fc2efc

**Comment by sakluger** [comment]
@rayane-djouah any luck reproducing? And just to confirm, when you remove the state tag, do you see the correct `tag missing` error?

**Comment by rezkiy37** [comment]
Well, I will try to reproduce it one more time.

**Comment by sakluger** [comment]
Thanks @rezkiy37. Sounds like we need to update the backend to support multiple errors for tags, is that right?

**Comment by cead22** [comment]
Looking at the php code that calculates violations, it looks like this is a case where we calculate a "tag out of policy" / "tag no longer valid" violation.

I think this can be improved, but I would suggest waiting until dependent tags are fully implemented in new dot. Curious if you agree @JmillsExpensify.

Right now, dependent tags aren't fully implemented in new dot, in the sense that you can have dependent tags but you can still select a child tag that doesn't correspond to the selected parent tag. As you can see in the example below, Eugene's parent tag is South, but I can still select it while having selected California for State and North for Region

<img width="336" alt="image" src="https://github.com/user-attachments/assets/2d1773c2-a91f-4c9a-b8aa-31e385608a83">

**Comment by sakluger** [comment]
Good catch @cead22. Seems like we should close this bug until we've fully built out support for dependent tags. I'll wait for @JmillsExpensify's feedback before closing.

**Comment by JmillsExpensify** [comment]
Yes, I agree with Carlos. Let's put this on hold for @yuwenmemon's dependent tag project.

**Comment by sakluger** [comment]
@yuwenmemon is there a particular GH issue that I can put this on hold for? Or could we close it because you will naturally build this as part of your dependent tag project?

**Comment by yuwenmemon** [comment]
Let's put this on HOLD waiting on https://github.com/Expensify/Expensify/issues/418655

And then we can include it as part of the polish issues for that project (there's a chance we just fix it as part of implementation)

**Comment by sakluger** [comment]
Sounds good! I'm adding this to the `#expense` project to line up with the issue it's on hold for: https://github.com/Expensify/Expensify/issues/418655

**Comment by sakluger** [comment]
The hold issue is also still on hold.

**Comment by sakluger** [comment]
The hold issue (https://github.com/Expensify/Expensify/issues/418655) is still on hold.

**Comment by rezkiy37** [comment]
@sakluger, any updates? 

More context - https://github.com/Expensify/App/issues/49504#issuecomment-2578806748.

**Comment by rezkiy37** [comment]
I’m planning a short vacation next week from Monday to Wednesday, from the 17th to the 19th. I am working on the issue later.

**Comment by rezkiy37** [comment]
I am back.

**Comment by sakluger** [comment]
The Multi-tag configuration / Dependent tags in NewDot project ([GH](https://github.com/Expensify/Expensify/issues/418655)) has been taken off hold! It's still a major feature that needs to go through the standard design and development process, so monthly is still appropriate for this issue.

**Comment by rezkiy37** [comment]
I will start work on the issue soon.

**Comment by rezkiy37** [comment]
@sakluger, can you update the issue's title to remove `[HOLD]`? Thank you!

**Comment by sakluger** [comment]
Hold is removed! I'm OOO for the next two weeks - if you need any changes made. please request another BZ team member to be added.

**Comment by sakluger** [comment]
@rezkiy37 any updates? Should we change the priority back to Weekly?

**Comment by VickyStash** [comment]
Hey @sakluger 👋 
@rezkiy37 doesn't work on Expensify anymore, so please unassign him from this issue.
I've checked this issue comments and I see the source of the bug was on the BE: https://github.com/Expensify/App/issues/49504#issuecomment-2400766744
So I believe this one should be internal, right?

**Comment by sakluger** [comment]
Hey @VickyStash, thanks for the reminder. I've updated the assignments.

I read through the issue again, and I think I removed [HOLD] from the title prematurely. Even though the [hold issue](https://github.com/Expensify/Expensify/issues/418655) was taken off hold, it is still in development. I'm going to put this one back on hold as we wait for that project to be completed. Once complete, we can retest (and hopefully close).

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
@kavimuru that's really strange! I check `agexptest+rt1308@gmail.com`'s account (the one in the second recording) on MacOS Chrome, and was able to load the Workspace Tags settings page without issue.

@yuwenmemon - there appears to be a bug on Android that throws an error when the admin tries accessing the Workspace Tags settings page.
